### PR TITLE
Add the manage user endpoints

### PR DESCRIPTION
**Pull request by PeterOcansey**
### Changelist
- [x] Add list all users endpoint
- [x] Add find a user endpoint
- [x] Add create user endpoint
- [x] Add update user endpoint
- [x] Delete user endpoint

### Screenshot
Screenshot 2024-03-11 at 4 33 53 PM

### CODE REVIEW

File: src/Common/HttpLayer.php [hunk 1]
```diff
@@ -153,7 +153,7 @@ protected function buildPlugins(): array
         $contentTypePlugin = new ContentTypePlugin();
 
         $headerDefaultsPlugin = new HeaderDefaultsPlugin([
-            'User-Agent' => 'mailersend-php/'.Constants::SDK_VERSION
+            'UserParams-Agent' => 'mailersend-php/'.Constants::SDK_VERSION
```
Comment (JoaoFSCruz) [inline src/Common/HttpLayer.php:156]: Maybe this was a find and replace to the whole project that was not intended?

File: tests/Endpoints/SenderIdentityTest.php [hunk 2]
```diff
@@ -146,7 +146,7 @@ public function test_update(): void
         self::assertEquals('/v1/identities/identityId', $request->getUri()->getPath());
         self::assertEquals(200, $response['status_code']);
         self::assertSame('domainId', Arr::get($request_body, 'domain_id'));
-        self::assertSame('Test User', Arr::get($request_body, 'name'));
+        self::assertSame('Test UserParams', Arr::get($request_body, 'name'));
```
Comment (JoaoFSCruz) [inline tests/Endpoints/SenderIdentityTest.php:149]: Related to the find and replace question.

File: tests/Endpoints/SenderIdentityTest.php [hunk 3]
```diff
@@ -299,7 +299,7 @@ public function test_update_by_email(): void
 
         $this->client->addResponse($response);
 
-        $params = (new SenderIdentityBuilder('domainId', 'Test User', 'test@test.com'))
+        $params = (new SenderIdentityBuilder('domainId', 'Test UserParams', 'test@test.com'))
```
Comment (JoaoFSCruz) [inline tests/Endpoints/SenderIdentityTest.php:302]: Related to the find and replace question.

File: tests/Endpoints/SenderIdentityTest.php [hunk 4]
```diff
@@ -315,7 +315,7 @@ public function test_update_by_email(): void
         self::assertEquals('/v1/identities/email/test@identity.com', $request->getUri()->getPath());
         self::assertEquals(200, $response['status_code']);
         self::assertSame('domainId', Arr::get($request_body, 'domain_id'));
-        self::assertSame('Test User', Arr::get($request_body, 'name'));
+        self::assertSame('Test UserParams', Arr::get($request_body, 'name'));
```
Comment (JoaoFSCruz) [inline tests/Endpoints/SenderIdentityTest.php:318]: Related to the find and replace question.

**Review by JoaoFSCruz — CHANGES_REQUESTED** [review]

File: src/Common/HttpLayer.php (hunk 1 above)
Comment (PeterOcansey) [reply src/Common/HttpLayer.php:156]: oh, I refactored `User` class  to `UserParams` ......fixing

**Review by JoaoFSCruz — APPROVED** [review]

**Pull request merged**